zkevm-circuits/src/pi_circuit.rs

```diff
@@ -1456,13 +1456,40 @@ pub struct PiTestCircuit<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usi
 );
 
 #[cfg(any(feature = "test", test, feature = "test-circuits"))]
-impl<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usize>
-    PiTestCircuit<F, MAX_TXS, MAX_CALLDATA>
+impl<F: Field, const MAX_TXS: usize, const MAX_CALLDATA: usize> SubCircuit<F>
+    for PiTestCircuit<F, MAX_TXS, MAX_CALLDATA>
 {
+    type Config = PiCircuitConfig<F>;
+
+    fn new_from_block(block: &witness::Block<F>) -> Self {
+        let mut block = block.clone();
+        block.circuits_params.max_txs = MAX_TXS;
+        block.circuits_params.max_calldata = MAX_CALLDATA;
```

Comment:
```suggestion
        assert_eq!(block.circuits_params.max_txs, MAX_TXS);
        assert_eq!(block.circuits_params.max_calldata, MAX_CALLDATA);
```
We should always use the same values for circuit parameters in `circuit_params` and generic const of circuit types, so I think it's better to assert that this is true rather than overwrite the values, because it enforces us to follow the rule of using the same parameter values in types and values.  Would this be a problem in your scenario?
With this you could also skip the clone.